middlewares/metrics.go

```diff
@@ -19,13 +24,41 @@ type Metrics interface {
 // given Metrics implementation to expose and monitor Traefik metrics
 type MetricsWrapper struct {
 	Impl Metrics
+
+	//frontend/backend
+	typ     string
```

Comment:
I thought about naming it `type`, but it's reserved keyword. Phase sound good thought.